src/engine/strat_engine/filesystem.rs

```diff
@@ -150,19 +150,38 @@ impl StratFilesystem {
     }
 
     /// Get the mount_point for this filesystem
-    /// TODO Replace this code with something less brittle
     pub fn get_mount_point(&self) -> EngineResult<Option<PathBuf>> {
-        let output = Command::new("df")
-            .arg("--output=target")
-            .arg(&self.devnode())
-            .output()?;
-        if output.status.success() {
-            let output_str = String::from_utf8_lossy(&output.stdout);
-            return Ok(output_str.lines().last().map(PathBuf::from));
+        if let Some(search) = self.devnode().to_str() {
+            match MountIter::new_from_proc() {
+                Ok(m_iter) => {
+                    for mp in m_iter {
+                        match mp {
+                            Ok(mount) => {
+                                if mount.contains(&MountParam::Spec(search)) {
```

Comment:
I guess there couldn't possibly be two matches for search?

Reply:
There could be `N` matching items for search, but all we are trying to do is find one so we can call `xfs` expand passing it as one of the arguments.

Reply:
If any are equally good, then it is better not to be eager about returning an error if there was a parse failure on any entry.